include/zephyr/drivers/stepper.h

```diff
@@ -0,0 +1,353 @@
+/**
+ * @file
+ *
+ * @brief Public API for Stepper Motor Controller
+ *
+ */
+
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_STEPPER_H
+#define ZEPHYR_INCLUDE_STEPPER_H
+
+/**
+ * @brief Stepper Motor Controller Interface
+ * @defgroup stepper_interface Stepper Motor Controller Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/kernel.h>
+#include <errno.h>
```

Comment:
you're not using any kernel API here, please, include what is needed

Reply:
we are now using k_poll_signal as suggested by Bjarki, hence i have kept this header